### PR TITLE
Self deployment and version visibility

### PR DESCRIPTION
**Pull request by sihil**
Make self deployment work inside the AWS environment.

This is connected with two other PRs: https://github.com/guardian/machine-images/pull/95 and https://github.com/guardian/deploy-tools-platform/pull/10. 

This PR also makes the build ID visible again at the bottom right of each page:
image

### CODE REVIEW

File: project/MagentaBuild.scala [hunk 1]
```diff
@@ -41,4 +51,28 @@ object MagentaBuild extends Build {
   ) ++ net.virtualvoid.sbt.graph.Plugin.graphSettings
 
   val magentaVersion = "1.0"
+
+
+  implicit def string2Dequote(s: String): Object {val dequote: String} = new {
+    lazy val dequote = s.replace("\"", "")
+  }
+
+  def buildFile(outDir: File, branch: String, buildNum: String, vcsNum: String, s: TaskStreams) = {
```
Comment (philwills) [inline project/MagentaBuild.scala]: What's the reason for not using https://github.com/sbt/sbt-buildinfo?

File: magenta-lib/src/main/scala/magenta/deployment_type/SelfDeploy.scala [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  val bucket = Param[String]("bucket",
+    """
+      |S3 bucket name to upload artifact into.
+      |
+      |The path in the bucket is `<stack>/<stage>/<packageName>/<fileName>`.
+    """.stripMargin
+  )
+
+  val prefixStage = Param[Boolean]("prefixStage",
```
Comment (philwills) [inline magenta-lib/src/main/scala/magenta/deployment_type/SelfDeploy.scala]: We should probably try and cut-down on the c&p of these params some time.

**Comment by philwills** [comment]
:+1:

File: project/MagentaBuild.scala (hunk 1 above)
Comment (sihil) [inline project/MagentaBuild.scala]: I wasn't aware of it. Will have a look at refactoring to use that instead.

File: magenta-lib/src/main/scala/magenta/deployment_type/SelfDeploy.scala (hunk 2 above)
Comment (sihil) [inline magenta-lib/src/main/scala/magenta/deployment_type/SelfDeploy.scala]: When I was reviewing this I thought I might actually get rid of them entirely for this deploy type. I can't see why the values would ever need to be customised.

**Pull request merged**